openmmtools/alchemy.py

```diff
@@ -231,6 +233,9 @@ def from_system(cls, system, *args, **kwargs):
             If specified, the state will search for a modified
             version of the alchemical parameters with the name
             ``parameter_name + '_' + parameters_name_suffix``.
+        custom_lambdas : list of str, optional
```

Comment:
I'd prefer not to expose this argument. The reason is simply that dynamically-created classes do not behave nicely in many cases (serialization above all), which we'll have to support if we expose this feature. The encouraged way to extend this is through (static) class extension
```python
class MyAlchemicalState(AlchemicalState):
   custom_param1 = AlchemicalState._LambdaParameter(...)
   custom_param2 = AlchemicalState._LambdaParameter(...)
```
If your application requires the dynamic instantiation, could you wrap these lines in a function in your code? I don't think that a `MultiStateSampler` would currently be able to resume correctly if you'll pass one of these classes to it though.